src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php

```diff
@@ -117,6 +118,7 @@ public function build(ContainerBuilder $container)
         $container->addCompilerPass(new CachePoolPrunerPass(), PassConfig::TYPE_AFTER_REMOVING);
         $this->addCompilerPassIfExists($container, FormPass::class);
         $container->addCompilerPass(new WorkflowGuardListenerPass());
+        $container->addCompilerPass(new ResettableServicePass('kernel.reset'), PassConfig::TYPE_AFTER_REMOVING);
```

Comment:
doing this after removing is a bad idea, as some resettable services may get inlined (due to being referenced only once before this pass adds a second reference). It should stay before optimizations are performed.
the general rule is that anything *adding* references should never run after optimizations (as optimizations are applied based on references). Removing references is fine (it may make us miss some optimization opportunities, but it cannot break the container)

Reply:
This was a suggestion by @nicolas-grekas. I see your point and would revert this line, if it's okay with @nicolas-grekas.

Reply:
I'm OK yes, let's revert

Reply:
Done.